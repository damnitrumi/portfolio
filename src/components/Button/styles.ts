import styled, { css } from "styled-components";

export const Wrapper = styled.button`
    ${() => css`
      border: none;
      background-color: transparent;
      cursor: pointer;
    `}
`;
