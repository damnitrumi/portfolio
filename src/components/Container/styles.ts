import styled, { css } from "styled-components";

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.primaryColor};
        padding: ${theme.paddings.containerDefault};
        height: 100vh;

        @media ${theme.media.ltaMedium}{
            padding: ${theme.paddings.containerLtaMedium};
        }
    `}
`;
