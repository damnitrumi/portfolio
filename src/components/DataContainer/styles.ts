import styled, { css } from "styled-components";

export const Wrapper = styled.div`
    ${() => css`
        max-width: 1100px;
        margin: 0 auto;
        height: 100%;
        padding: 200px 0;
    `}
`;
