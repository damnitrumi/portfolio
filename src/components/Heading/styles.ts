import styled, { css } from "styled-components";
import { HeadingProps } from ".";

const pickSize = (size: string) => {
  if (size) {
    return css`
      font-size: ${size};
    `;
  }
};

export const Title = styled.h1<HeadingProps>`
${({ size, as }) => css`
  color: white;
  ${pickSize(size)}
  font-weight: ${as == "h1" ? 500 : 400};
`}
`;
