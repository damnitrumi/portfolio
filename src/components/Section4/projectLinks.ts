type ProjectLinksProps = {
  link: string;
  srcImg: string;
  name: string;
};

export const projectLinks: ProjectLinksProps[] = [
  {
    link: "https://netflix-clone-light-nextjs-damnitrumi.vercel.app/",
    srcImg: "/assets/images/netflix-light.jpg",
    name: "Netflix Recreation",
  },
  {
    link: "https://weather-app-nextjs-damnitrumi.vercel.app/",
    srcImg: "/assets/images/weather-app.jpg",
    name: "Weather App",
  },
  {
    link: "https://to-do-list-damnitrumi.vercel.app/",
    srcImg: "/assets/images/to-do-list.jpg",
    name: "To-do List",
  },
  {
    link: "https://digidex-react-js-damnitrumi.vercel.app/",
    srcImg: "/assets/images/digidex.jpg",
    name: "Digidex",
  },
  {
    link: "https://consulta-cep-react-js-damnitrumi.vercel.app/",
    srcImg: "/assets/images/consulta-cep.jpg",
    name: "Consulta CEP",
  },
  {
    link: "https://posts-page-react-js-damnitrumi.vercel.app/",
    srcImg: "/assets/images/posts-page.jpg",
    name: "Posts Page",
  },
  {
    link: "https://monstercat-recreation-damnitrumi.vercel.app/",
    srcImg: "/assets/images/monstercat-recreation.jpg",
    name: "Monstercat Recreation",
  },
  {
    link: "https://ableton-recreation-damnitrumi.vercel.app/",
    srcImg: "/assets/images/ableton-recreation.jpg",
    name: "Ableton Recreation",
  },
  {
    link: "https://calculadora-de-imc-damnitrumi.vercel.app/",
    srcImg: "/assets/images/imc-calc.jpg",
    name: "Calculadora de IMC",
  },
  {
    link: "https://wildbeast-origamid-damnitrumi.vercel.app/",
    srcImg: "/assets/images/wildbeast.jpg",
    name: "WildBeast - Origamid",
  },
  {
    link: "https://flexblog-origamid-damnitrumi.vercel.app/",
    srcImg: "/assets/images/flexblog.jpg",
    name: "FlexBlog - Origamid",
  },
];
