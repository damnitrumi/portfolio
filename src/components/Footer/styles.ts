import styled, { css } from "styled-components";
import { Wrapper as Container } from "../Container/styles";

export const Wrapper = styled.div`
  ${() => css`
  `}
`;

export const FooterContainer = styled(Container)`
    background-color: transparent;
    height: auto;
    position: fixed;
    width: 100%;
    bottom: 0;
`;

export const LinksContainer = styled.div``;
