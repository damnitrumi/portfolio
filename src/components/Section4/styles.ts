import styled, { css } from "styled-components";
import { Wrapper as DataContainer } from "../DataContainer/styles";

export const Wrapper = styled.section`
  ${DataContainer}{
    text-align: center;
  }
`;

export const WebsiteContainer = styled.div`
    ${() => css`
      display: grid;
      gap: 10px;
      grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
      overflow-y: auto;
      &::-webkit-scrollbar{
        display: none;
      }
      margin-top: 20px;
    `}
`;
