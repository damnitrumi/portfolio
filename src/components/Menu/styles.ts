import styled, { css } from "styled-components";
import { MenuBehaviorProps } from ".";
import { LinksContainer as IconsContainerMock } from "../Footer/styles";
import { StyledIconBase } from "@styled-icons/styled-icon";

export const Wrapper = styled.div<MenuBehaviorProps>`
    ${({ theme, menuVisible }) => css`
    position: fixed;
    width: 100%;
    max-width: 500px;
    top: 0;
    right: -100%;
    transition: all .8s ease-in-out;

    @media ${theme.media.ltaMedium} {
      max-width: 100%;
    };

    ${
      menuVisible &&
      css`
        right: 0;
    `
    }
    `}
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;
export const LinksContainer = styled.div`
  padding: 30px 0;

  a{
    padding: 20px 0;
    color: ${({ theme }) => theme.colors.menuGray};
    transition: all 200ms ease-in;
  }

  a:hover{
    color: ${({ theme }) => theme.colors.white};
  }
`;

export const IconsContainer = styled(IconsContainerMock)`
  ${StyledIconBase}{
    color: ${({ theme }) => theme.colors.menuGray};
    &:hover{
      color: ${({ theme }) => theme.colors.hoverGreen};
    }
  }
`;
