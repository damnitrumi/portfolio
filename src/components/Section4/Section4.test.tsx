import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Section4 } from ".";
import { projectLinks } from "./projectLinks";

describe("<Section4 />", () => {
  it("should render Section4", () => {
    renderTheme(<Section4 />);

    expect(
      screen.getByRole("heading", { name: "Projetos" }),
    ).toBeInTheDocument();

    expect(screen.getAllByRole("link")).toHaveLength(projectLinks.length);
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<Section4 />);

    expect(container).toMatchSnapshot();
  });
});
