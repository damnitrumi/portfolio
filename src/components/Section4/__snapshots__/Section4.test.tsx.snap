// Vitest Snapshot v1

exports[`<Section4 /> > should match snapshot 1`] = `
.c1 {
  background-color: #1b1b1b;
  padding: 30px 60px;
  height: 100vh;
}

.c3 {
  max-width: 1100px;
  margin: 0 auto;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  color: #FFFFFF;
  font-size: 3.8rem;
  font-weight: 400;
}

.c9 {
  color: #FFFFFF;
  font-size: 1.6rem;
  line-height: 30px;
}

.c6 {
  text-align: center;
}

.c6 .c8 {
  -webkit-transition: all 200ms ease-in;
  transition: all 200ms ease-in;
}

.c6:hover .c8 {
  color: #06c819;
}

.c7 {
  overflow: hidden;
  height: 200px;
}

.c7 img {
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
  height: 100%;
  object-fit: cover;
}

.c7:hover img {
  -webkit-transform: scale(1.5) rotate(15deg);
  -ms-transform: scale(1.5) rotate(15deg);
  transform: scale(1.5) rotate(15deg);
}

.c0 .c2 {
  text-align: center;
}

.c5 {
  display: grid;
  gap: 10px;
  grid-template-columns: repeat(auto-fit,minmax(300px,1fr));
  overflow-y: auto;
  margin-top: 30px;
}

.c5::-webkit-scrollbar {
  display: none;
}

@media (max-width:768px) {
  .c1 {
    padding: 20px;
  }
}

@media (max-width:768px) {
  .c7 {
    height: 300px;
  }
}

<div>
  <section
    class="c0"
    id="projetos"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <h5
          class="c4"
        >
          Projetos
        </h5>
        <div
          class="c5"
        >
          <a
            class="c6"
            href="https://digidex-react-js-damnitrumi.vercel.app/"
            target="_blank"
          >
            <div
              class="c7"
            >
              <img
                alt="Digidex"
                src="/assets/images/digidex.jpg"
              />
            </div>
            <p
              class="c8 c9"
            >
              Digidex
            </p>
          </a>
          <a
            class="c6"
            href="https://consulta-cep-react-js-damnitrumi.vercel.app/"
            target="_blank"
          >
            <div
              class="c7"
            >
              <img
                alt="Consulta CEP"
                src="/assets/images/consulta-cep.jpg"
              />
            </div>
            <p
              class="c8 c9"
            >
              Consulta CEP
            </p>
          </a>
          <a
            class="c6"
            href="https://monstercat-recreation-damnitrumi.vercel.app/"
            target="_blank"
          >
            <div
              class="c7"
            >
              <img
                alt="Monstercat Recreation"
                src="/assets/images/monstercat-recreation.jpg"
              />
            </div>
            <p
              class="c8 c9"
            >
              Monstercat Recreation
            </p>
          </a>
          <a
            class="c6"
            href="https://ableton-recreation-damnitrumi.vercel.app/"
            target="_blank"
          >
            <div
              class="c7"
            >
              <img
                alt="Ableton Recreation"
                src="/assets/images/ableton-recreation.jpg"
              />
            </div>
            <p
              class="c8 c9"
            >
              Ableton Recreation
            </p>
          </a>
          <a
            class="c6"
            href="https://calculadora-de-imc-damnitrumi.vercel.app/"
            target="_blank"
          >
            <div
              class="c7"
            >
              <img
                alt="Calculadora de IMC"
                src="/assets/images/imc-calc.jpg"
              />
            </div>
            <p
              class="c8 c9"
            >
              Calculadora de IMC
            </p>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
